Replace jaccard_similarity_score with jaccard_score

joanne-tsai opened the pull request:
jaccard_similarity_score was deprecated since version 0.21 by sklearn

joanne-tsai committed:
Replace jaccard_similarity_score with jaccard_score

jaccard_similarity_score was deprecated since version 0.21 by sklearn

joanne-tsai committed:
Fix jaccard_score

jaccard_score defaults to calculating the binary average, but labels were
multilabel. This commit changes the average to micro. Not sure if it is the
right average, but it matches the assert.

brianhelba commented on tests/test_metrics_comparative.py:
```diff
@@ -29,12 +29,10 @@ def test_binary_jaccard_reference(truth_binary_values, prediction_binary_values)
-    # see: https://github.com/scikit-learn/scikit-learn/issues/3037
-    reference_value = sklearn.metrics.jaccard_similarity_score(
+    reference_value = sklearn.metrics.jaccard_score(
         np.expand_dims(truth_binary_values, axis=0),
```
Do we still need to wrap these with [`np.expand_dims`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expand_dims.html)? That was previously taking an array like `[1, 0, 1]` and wrapping it into `[[1, 0, 1]]`. Is the new implementation of `sklearn.metrics.jaccard_score` now able to take the arrays directly (like all the other `sklearn.metrics`)?

brianhelba commented on tests/test_metrics_comparative.py:
```diff
@@ -32,6 +32,7 @@ def test_binary_jaccard_reference(truth_binary_values, prediction_binary_values)
     reference_value = sklearn.metrics.jaccard_score(
         np.expand_dims(truth_binary_values, axis=0),
         np.expand_dims(prediction_binary_values, axis=0),
+        average='micro',
```
Since this is a binary classification, we shouldn't need to think about the averaging strategy. For what it's worth, you can read up on micro vs macro here: https://datascience.stackexchange.com/a/16001 . Note, most of our "aggregrate" metrics use a macro averaging strategy, except for the balanced multiclass accuracy, which uses micro averaging (the *micro* is what makes it *balanced*).

In this case, does the need to use `micro` go away if we get rid of the extra array wrapping?

joanne-tsai committed:
Fix jaccard_score call

np.expand_dims(...) made jaccard_score(...) think that it was multilabel
when it wasn't. np.expand_dims(...) was necessary for the old jaccard
function, but is no longer needed.

brianhelba reviewed (approved):
Great!
